setup.py

```diff
@@ -1,13 +1,21 @@
+import subprocess
+import sys
 from distutils.core import setup
-from subprocess import call
+
+
+DATA_FILES = []
 
 
 def generate_requirements():
     with open('requirements.txt', 'w') as f:
-        call(['pipenv', 'lock', '--requirements'], stdout=f)
+        subprocess.check_call(['pipenv', 'lock', '--requirements'], stdout=f)
     return 'requirements.txt'
 
 
+if 'sdist' in sys.argv:
+    DATA_FILES.append(('requirements.txt', generate_requirements()))
```

Comment:
Returning "requirements.txt" from generate_requirements() is little hacky. generate_requirements should not have a return value, we call it for the side effect.

If we need to add the tuple ("requirements.txt", "requirements.txt") specifying it as is it simpler and more clear.

I think the correct way to do this is to create our own sdist class, and create the generated files before we run the real sdist.

Basically this is something like:
```
from distutils.command import sdist


class rose_sdist(sdist.sdist):

    def run(self):
        self.genrate_files()
        sdist.sdist.run(self)

    def generate_files(self):
        # generate the requirements file here...

detup(
    cmdclass = {"sdist": "rose_sdist"},
    ...
)
```
See how this is done for overriding the install_data class in https://wiki.python.org/moin/Distutils/Tutorial.

We can also generate the file as part of the "build" command - doesn't "sdist" depend on "build"?